backend/src/data/repositories/course/course.repository.ts

```diff
@@ -73,6 +75,45 @@ class Course {
     };
   }
 
+  public async getAllCoursesStudying(
+    userId: number,
+  ): Promise<CourseGetResponseDto[]> {
+    return this.#CourseModel
+      .query()
+      .select(
+        'courses.id',
+        'title',
+        'description',
+        'url',
+        'imageUrl',
+        'courseCategoryId',
+      )
+      .distinct('courseId')
+      .withGraphJoined('[mentees, category, vendor]')
+      .where('menteeId', userId)
+      .whereNotNull('mentorId')
```

Comment:
@fshabanov now in the db mentorId could be null, so basically user can not have a mentor assigned
and this condition is written in the trello card - show only those studying courses, on which user already has a mentor

Reply:
Even if the mentorId can be null in db, if the user has no mentor assigned, that means user is not a mentee for the course, so, basically, `.where('menteeId', userId)` will not return a user and there's no need for `.whereNotNull(...)`
I mean if the user is a mentee for a course, that means this user for sure has a mentor and `mentorId` cannot be null

Reply:
@fshabanov generally your point sounds logical, cause `mentorToMentee` connection on the base flow with mentee to course assignment requires `mentorId`,, but as in db this col is defined as `nullable` - then I think this additional `.whereNotNull` check won't be redundant - just to avoid some edge cases